src/libraries/System.Linq/src/System/Linq/Reverse.cs

```diff
@@ -23,6 +23,21 @@ public static IEnumerable<TSource> Reverse<TSource>(this IEnumerable<TSource> so
             return new ReverseIterator<TSource>(source);
         }
 
+        public static IEnumerable<TSource> Reverse<TSource>(this T[] source)
```

Comment:
```suggestion
        public static IEnumerable<TSource> Reverse<TSource>(this TSource[] source)
```
I did this _exact_ same thing in the API proposal...